app/src/main/kotlin/hr/asee/android/template/compose/ui/prelogin/register/RegisterScreen.kt

```diff
@@ -0,0 +1,301 @@
+                keyboardOptions = KeyboardOptions(
+                    keyboardType = KeyboardType.Text,
+                    imeAction = ImeAction.Next),
+                keyboardActions = KeyboardActions(
+                    onNext = { focusManager.moveFocus(FocusDirection.Down)}
+
+                )
+            )
+            if(isEmailValid()){
+                Text(
+                    text = stringResource(R.string.register_screen_email_not_correct_text_label),
+                    color = AssecoYellow,
+                    fontSize = 12.sp,
+                    fontFamily = Geomanist,
+                    modifier = Modifier.padding(start = 8.dp)
+                )
+            }
+        }
+
+        //PASSWORD FIELD
+
+        var passwordVisible by rememberSaveable { mutableStateOf(false) }
+
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.Start
+        ){
+            InputField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(
+                        shape = RoundedCornerShape(20),
+                        color = Color.Transparent
+                    )
+                    .border(width = 1.dp, color = Color.Black, shape = RoundedCornerShape(20)),
+                state = passwordState,
+                label = stringResource(R.string.register_screen_password_field_label),
+                keyboardOptions = KeyboardOptions(
+                    keyboardType = KeyboardType.Text,
+                    imeAction = ImeAction.Next),
+                keyboardActions = KeyboardActions(
+                    onNext = { focusManager.moveFocus(FocusDirection.Down)}
+                ),
+                visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
+                trailingIconContent = {
+                    TextVisibilityIcon(
+                        isVisible = passwordVisible,
+                        onVisibilityClicked = { passwordVisible = !passwordVisible },
+                        tint = AssecoBlue
+                    )
+                }
+            )
+            if (isPasswordValid()) {
+                Text(
+                    text = stringResource(R.string.register_screen_password_not_correct_text_label),
+                    color = AssecoYellow,
+                    fontSize = 12.sp,
+                    fontFamily = Geomanist,
+                    modifier = Modifier.padding(start = 8.dp)
+                )
+            }
+            else if(isPasswordLongEnough()){
+                Text(
+                    text = stringResource(R.string.register_screen_password_too_short_text_label),
+                    color = AssecoYellow,
+                    fontSize = 12.sp,
+                    fontFamily = Geomanist,
+                    modifier = Modifier.padding(start = 8.dp)
+                )
+            }
+        }
```

Comment:
Ne trebaju dvije provjere za ovo, dovoljno je jednu, `isValid` pa tamo provjeravati i jel dovoljno duga lozinka i dodati to u tekst greške